zerver/tests/test_upload.py

```diff
@@ -416,6 +416,37 @@ def test_multiple_claim_attachments_different_owners(self) -> None:
         self.assertEqual(Attachment.objects.get(path_id=d1_path_id).messages.count(), 3)
         self.assertTrue(Attachment.objects.get(path_id=d1_path_id).is_realm_public)
 
+    def test_web_public_attachment_claim(self) -> None:
+        self.login('hamlet')
+        d1 = StringIO("zulip!")
+        d1.name = "dummy_1.txt"
+        result = self.client_post("/json/user_uploads", {'file': d1})
+        d1_path_id = re.sub('/user_uploads/', '', result.json()['uri'])
+        host = self.example_user('hamlet').realm.host
+        attachment = Attachment.objects.get(path_id=d1_path_id)
+        self.assertFalse(attachment.is_web_public)
+
+        self.subscribe(self.example_user("hamlet"), "Rome")
+
+        # Send to the web-public stream, Rome.
+        body = f"First message ...[zulip.txt](http://{host}/user_uploads/" + d1_path_id + ")"
+        self.send_stream_message(self.example_user("hamlet"), "Venice", body, "test")
+        attachment = Attachment.objects.get(path_id=d1_path_id)
+        self.assertTrue(attachment.is_realm_public)
+        self.assertFalse(attachment.is_web_public)
+
+        body = f"First message ...[zulip.txt](http://{host}/user_uploads/" + d1_path_id + ")"
+        self.send_stream_message(self.example_user("hamlet"), "Denmark", body, "test")
+        attachment = Attachment.objects.get(path_id=d1_path_id)
+        self.assertTrue(attachment.is_realm_public)
+        self.assertFalse(attachment.is_web_public)
+
+        body = f"First message ...[zulip.txt](http://{host}/user_uploads/" + d1_path_id + ")"
+        self.send_stream_message(self.example_user("hamlet"), "Denmark", body, "test")
+        attachment = Attachment.objects.get(path_id=d1_path_id)
+        self.assertTrue(attachment.is_realm_public)
+        self.assertFalse(attachment.is_web_public)
```

Comment:
This test seems to have been copy-pasted and then not finished; you don't ever send to `Rome`.

Reply:
I'm just going to rewrite this to extend the previous test in the natural way.